compiler/rustc_const_eval/src/const_eval/valtrees.rs

```diff
@@ -236,18 +247,26 @@ pub(crate) fn eval_to_valtree<'tcx>(
     let const_alloc = tcx.eval_to_allocation_raw(param_env.and(cid))?;
 
     // FIXME Need to provide a span to `eval_to_valtree`
-    let ecx = mk_eval_cx_to_read_const_val(
+    let mut ecx = mk_eval_cx_to_read_const_val(
         tcx,
         DUMMY_SP,
         param_env,
         // It is absolutely crucial for soundness that
         // we do not read from mutable memory.
-        CanAccessMutGlobal::No,
+        GlobalAccessPermissions::Static,
     );
     let place = ecx.raw_const_to_mplace(const_alloc).unwrap();
     debug!(?place);
 
     let mut num_nodes = 0;
+
+    // To preserve provenance of static items, it is crucial that we do not
```

Comment:
I'd say this is less about provenance and more about preserving their unique absolute address.